nsxt/resource_nsxt_edge_cluster.go

```diff
@@ -0,0 +1,300 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+var deploymentTypeValues = []string{
+	model.EdgeCluster_DEPLOYMENT_TYPE_VIRTUAL_MACHINE,
+	model.EdgeCluster_DEPLOYMENT_TYPE_PHYSICAL_MACHINE,
+	model.EdgeCluster_DEPLOYMENT_TYPE_UNKNOWN,
+}
+
+var memberNodeTypeValues = []string{
+	model.EdgeCluster_MEMBER_NODE_TYPE_EDGE_NODE,
+	model.EdgeCluster_MEMBER_NODE_TYPE_PUBLIC_CLOUD_GATEWAY_NODE,
+	model.EdgeCluster_MEMBER_NODE_TYPE_UNKNOWN,
+}
+
+func resourceNsxtEdgeCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtEdgeClusterCreate,
+		Read:   resourceNsxtEdgeClusterRead,
+		Update: resourceNsxtEdgeClusterUpdate,
+		Delete: resourceNsxtEdgeClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+			},
+			"tag": getTagsSchema(),
+			"cluster_profile_ids": {
+				Type:        schema.TypeList,
+				Description: "Edge cluster profile Ids",
+				Optional:    true,
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"deployment_type": {
+				Type:        schema.TypeString,
+				Description: "Edge cluster deployment type",
+				Computed:    true,
```

Comment:
Do those flags have a default?

Reply:
As a convention, do we apply the NSX defaults in the provider? Or let NSX apply them as we pass null?

Reply:
BTW deployment_type is readonly: should we even include this in the schema? This is in general, not specific to deployment_type

Reply:
We generally try to avoid `Computed` attributes wherever possible, so yes, we assign default on the provider side

Reply:
If we think the attribute can be useful to the user, we can add it as `Computed`

Reply:
As I don't see any special use for this, I'll remove this attribute entirely